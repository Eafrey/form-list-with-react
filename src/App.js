import React, { Component } from 'react';
import './App.css';
import Modal from "antd/lib/modal";
import Radio from "antd/lib/radio";
import RadioGroup from "antd/lib/radio/group";

// import _ from "lodash"

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showDelete: true,
      preview: 'preview',
      dialogShow: false,
      selectRadio: 0,
      list: [],
      delete: 'delete'
    };
  }

  showDialog = () => {
    this.setState({ dialogShow: true });
  };
  handleCancelDialog = () => {
    this.setState({ dialogShow: false });
  };

  onChange = e => {
    this.setState({ selectRadio: e.target.value });
  };
  onPush = () => {
    let list = this.state.list;
    list.push({ type: this.state.selectRadio });
    this.setState({ list: list });
    this.setState({ dialogShow: false });
  };

  delete = (index) => {
    let list = this.state.list.filter((item, i) => i !== index);
    this.setState({ list: list });
  }

  preview = () => {
    if (this.state.preview === 'preview') {
      this.setState({ preview: 'edit' });
      this.setState({ showDelete: false });
      this.setState({ delete: "" });

      this.setState({ showDelete: false });
    } else if (this.state.preview === 'edit') {
      this.setState({ preview: 'preview' });
      this.setState({ showDelete: true });
      this.setState({ delete: "delete" });


    }
  };

  remove = index => {
    let list = this.state.list;
    this.state.list.splice(index, 1);

    // let list = this.state.list.filter((item, i) => i !== index);
    this.setState({ list: list });
  };

  add() {

    var name;
    name = prompt("请问你叫什么名字?");
    this.setState({
      add: "add"
    })

    console.log('nname' + name);

    const is = [];
    console.log('iiss' + is);
    is.push(
      <tr >
        <td>description</td>
        <td>url:</td>
      </tr>
      // <tr key={name} >
      //   <input type="text" />
      //   <button onClick={this.delete}>delete</button>
      // </tr>
    );
    console.log('iiss' + is);

    this.setState({
      infos: is
    });
  }


  render() {
    return (
      <div className="App">
        <button type="primary" onClick={this.preview}>
          {this.state.preview}
        </button>
        <button type="primary" onClick={this.showDialog}>
          add
        </button>
        <div style={{ width: 300 }}>
          {this.state.list.map((item, index) => {
            if (item.type === 1) {
              if(this.state.showDelete) {
              return <div><input type='text' placeholder="text" />
                <button type="primary" onClick={() => {
                  this.remove(index);
                }}>{this.state.delete}</button></div>;
              } else {
                return <div><input type='text' placeholder="text" /></div>;
              }
            } else {
              if(this.state.showDelete) {
                return <div><input type='date' />
                <button type="primary" onClick={() => {
                  this.remove(index);
                }}>{this.state.delete}</button></div>;
              } else {
                return <div><input type='date' /></div>; 
              }
              
            }

          })}
        </div>
        <Modal
          title="Select input type"
          visible={this.state.dialogShow}
          onOk={this.onPush}
          onCancel={this.handleCancelDialog}
        >
          <RadioGroup onChange={this.onChange} value={this.state.selectRadio}>
            <Radio value={1}>
              Text Input
            </Radio>
            <Radio value={2}>
              DatePicker
            </Radio>
          </RadioGroup>
        </Modal>
      </div>
    );
  }

}

export default App;

