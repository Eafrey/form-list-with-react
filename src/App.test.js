import React from 'react';
import ReactDOM from 'react-dom';
import renderer from "react-test-renderer"
import App from './App';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from "enzyme-adapter-react-16"
import { resolve } from 'path';

// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });

// it('renders App correctly', () => {
//   const tree = renderer.create(<App hello="world 2018" />).toJSON();
//   expect(tree).toMatchSnapshot();
// });

// it('check button1 state', () => {
//   Enzyme.configure({ adapter: new Adapter() });
//   const app = shallow(<App />);
//   expect(app.find('.App-intro').text()).toEqual('react demo');

//   app.find('button').at(0).simulate('click');

//   expect(app.find('.App-intro').text()).toEqual('react demo 2018 desc');


// })

// it('check button2 state,use state', () => {
//   Enzyme.configure({ adapter: new Adapter() });
//   const app = mount(<App />);
//   expect(app.state().appName).toEqual('world 2018');

//   app.find('button').at(1).simulate('click');
//   expect(app.state().appName).toEqual("react reduc app name")
// })

// it('check button2 state,use mount', () => {
//   Enzyme.configure({ adapter: new Adapter() });
//   const app = mount(<App />);
//   expect(app.find('.App-title').text()).toEqual('world 2018');
//   app.find('button').at(1).simulate('click');
//   expect(app.find('.App-title').text()).toEqual("react reduc app name")
// })

describe('', () => {
  beforeEach(function () {
    Enzyme.configure({ adapter: new Adapter() });
    global.fetch = jest.fn().mockImplementation(() => Promise.resolve( {
      ok: true,
      json: () => {
        return {
          items: [{ 'description': 'description123' }]
        }
      }
    }
    ));

  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders App correctly', () => {
    const tree = renderer.create(<App hello="world 2018" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('check button1 state', () => {
    Enzyme.configure({ adapter: new Adapter() });
    const app = shallow(<App />);
    expect(app.find('.App-intro').text()).toEqual('react demo');

    app.find('button').at(0).simulate('click');

    expect(app.find('.App-intro').text()).toEqual('react demo 2018 desc');


  })

  it('check button2 state,use state', () => {
    Enzyme.configure({ adapter: new Adapter() });
    const app = mount(<App />);
    expect(app.state().appName).toEqual('world 2018');

    app.find('button').at(1).simulate('click');
    expect(app.state().appName).toEqual("react reduc app name")
  })

  it('check button2 state,use mount', () => {
    Enzyme.configure({ adapter: new Adapter() });
    const app = mount(<App />);
    expect(app.find('.App-title').text()).toEqual('world 2018');
    app.find('button').at(1).simulate('click');
    expect(app.find('.App-title').text()).toEqual("react reduc app name")
  })

  it('should hava description', function () {
    const app = shallow(<App />);
    return new Promise(resolve => setTimeout(resolve, 0)).then(() => {
      expect(app.state().infos).toEqual( [{"description": "description123"}]);
    });
  });
})