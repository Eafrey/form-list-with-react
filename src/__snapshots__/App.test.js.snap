// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      world 2018
    </h1>
  </header>
  <p
    className="App-intro"
  >
    react demo
  </p>
  <table>
    <tbody />
  </table>
  <button
    onClick={[Function]}
  >
    change desc
  </button>
  <button
    onClick={[Function]}
  >
    change appName
  </button>
</div>
`;
